### PR TITLE
Part2

### PR DESCRIPTION
**Pull request by jukent**
.rst text for Part 2 of the tutorial

### CODE REVIEW

File: source/tutorials/yourfirst.rst [hunk 1]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+
+8. Let's learn more about the math module!
+
+   Hopefully since you already imported code from your :code:`readdata`\, :code:`printing`, and :code:computation` modules, importing from a built-in package :code:`math` seemed a little less intimidating.
+
+   So far you have only used the :code:`math.log` function, but let's test out some other common methods within :code:`math`\.
+
+   Perhaps you want to change the base of your logarithm. To do this you could type :code:`math.log(x, base)`. Here :code:`base` is
+   a keyword argument (just like :code:`filename` or :code:`types` in our :code:`read_data()` function) which means that :code:`base` does not need to be specified- 
+   when it is not the logarithm is assumed to be natural (base *e*). When both arguments are entered, the function returns the logarithm of :code:`x` to the given :code:`base`\,
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2309]:
```suggestion
   a keyword argument (just like :code:`filename` or :code:`types` in our :code:`read_data()` function) which
   means that :code:`base` does not need to be specified.  When it is not specified, the logarithm is assumed to
   be natural (base *e*). When both arguments are entered, the function returns the logarithm of :code:`x` to
   the given :code:`base`\, 
```

File: source/tutorials/yourfirst.rst [hunk 2]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+   Perhaps you want to change the base of your logarithm. To do this you could type :code:`math.log(x, base)`. Here :code:`base` is
+   a keyword argument (just like :code:`filename` or :code:`types` in our :code:`read_data()` function) which means that :code:`base` does not need to be specified- 
+   when it is not the logarithm is assumed to be natural (base *e*). When both arguments are entered, the function returns the logarithm of :code:`x` to the given :code:`base`\, 
+   calculated by :code:`log(x)/log(base)`\. Let's test this out:
+
+   .. code-block:: python
+
+      import math as m
+
+      x = math.e
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2316]:
```suggestion
      x = m.e
```

File: source/tutorials/yourfirst.rst [hunk 3]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+
+      y_natural = m.log(x)
+      y_base10 = m.log(x, 10)
+
+      print(x, y_natural, y_base10)
+
+   ..
+
+   Something new that we have done here is use the :code:`import ... as` statements. This essentially allows us to
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2325]:
```suggestion
   Something new that we have done here is use the \":code:`import ... as ...`\" statement. This essentially allows us to 
```

File: source/tutorials/yourfirst.rst [hunk 4]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+      y_base10 = m.log(x, 10)
+
+      print(x, y_natural, y_base10)
+
+   ..
+
+   Something new that we have done here is use the :code:`import ... as` statements. This essentially allows us to 
+   shorten the name of the module for convenience if it is very long or if we are going to be calling it a lot.
+
+   :code:`math.e` calls up Euler's number (*e*), the base of the natural logarithm. *e* is an irrational number with infinite decimal places, often approximated as 2.718.
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2328]:
```suggestion
   The symbol :code:`math.e` represents Euler's number (*e*), the base of the natural logarithm. Euler's number (*e*) is an irrational number with infinite decimal places, often approximated as 2.718. 
```

File: source/tutorials/yourfirst.rst [hunk 5]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+
+   ..
+
+   Something new that we have done here is use the :code:`import ... as` statements. This essentially allows us to 
+   shorten the name of the module for convenience if it is very long or if we are going to be calling it a lot.
+
+   :code:`math.e` calls up Euler's number (*e*), the base of the natural logarithm. *e* is an irrational number with infinite decimal places, often approximated as 2.718. 
+   How much more accurate is :code:`math.e` than this approximation?
+   
+   The above function :code:`math.log(x, base)` is very useful for computing logarithms in any base - but for some common bases there are separate logarithmic functions.
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2331]:
```suggestion
   The function :code:`math.log(x, base)` is very useful for computing logarithms in any base - but
   for some common bases there are separate logarithmic functions. 
```

File: source/tutorials/yourfirst.rst [hunk 6]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+   Something new that we have done here is use the :code:`import ... as` statements. This essentially allows us to 
+   shorten the name of the module for convenience if it is very long or if we are going to be calling it a lot.
+
+   :code:`math.e` calls up Euler's number (*e*), the base of the natural logarithm. *e* is an irrational number with infinite decimal places, often approximated as 2.718. 
+   How much more accurate is :code:`math.e` than this approximation?
+   
+   The above function :code:`math.log(x, base)` is very useful for computing logarithms in any base - but for some common bases there are separate logarithmic functions. 
+   Try using :code:`log10(x)`\:
+
+
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2334]:
```suggestion

```

File: source/tutorials/yourfirst.rst [hunk 7]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+      y_natural = m.log(x)
+      y_base10 = m.log(x, 10)
+      y_log10 = m.log10(x)
+
+      print(x, y_natural, y_base10, y_log10)
+
+   ..
+
+   Do the two values differ? :code:`math.log10(x)` is considered to be more accurate than :code:`math.log(x, 10)`\.
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2349]:
```suggestion
   Do the two values differ? The :code:`math.log10(x)` function is considered to be more accurate than :code:`math.log(x, 10)`\.
```

File: source/tutorials/yourfirst.rst [hunk 8]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+   Similarly :code:`math.log2(x)` is more accurate than :code:`math.log(x, 2)`\.
+
+9. Let's cover some :code:`math` trigonometry examples!
+
+   Pi is an irrational number (like *e*) that is approximately :math:`\frac{22}{7}` or 3.14159.
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2354]:
```suggestion
   The math symbol :math:`\pi` is an irrational number (like *e*) that is approximately :math:`\frac{22}{7}` or 3.14159. 
```

File: source/tutorials/yourfirst.rst [hunk 9]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+   Say we wanted to convert a number from 60 degrees to radians. We have two options:
+
+   .. code-block:: python
+
+      import math as m
+
+      deg = 60
+
+      rads = deg * m.pi / 180
+      rads_fromfunc = radians(deg)
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2366]:
```suggestion
      rads_fromfunc = m.radians(deg)
```

File: source/tutorials/yourfirst.rst [hunk 10]
```diff
@@ -1399,3 +1399,1012 @@ repository to GitHub.
...
+
+      print(deg, m.pi, ras, rads_fromfunc)
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2368]:
```suggestion
      print(deg, m.pi, rads, rads_fromfunc)
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
There are some typos.  It took a while to review, but I think I got most of them.

**Comment by jukent** [comment]
> There are some typos. It took a while to review, but I think I got most of them.

Thanks! There were a lot of typos - I'm sorry for that. It is a rather large document.

**Comment by kmpaul** [comment]
Yeah. I understand, though. 

This could be a reason to move to a different theme. Then we could break it up into multiple pages.

**Comment by kmpaul** [comment]
I'll see if I can review this today.

File: source/tutorials/yourfirst.rst [hunk 11]
```diff
@@ -1399,3 +1399,1020 @@ repository to GitHub.
...
+
+    or
+
+    .. code-block:: python
+       :lineno-start: 37
+
+       # Output comparison of data
+       print_comparison('HEAT INDX', data['date'], data['time'], data['heatindex'], heatindex)
+
+    ..
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:1854]:
```suggestion
       .. code-block:: python
          :lineno-start: 29

          # Output comparison of data
          print_comparison('WINDCHILL', data['date'], data['time'], data['windchill'], windchill)

       ..

       or

       .. code-block:: python
          :lineno-start: 37

          # Output comparison of data
          print_comparison('HEAT INDX', data['date'], data['time'], data['heatindex'], heatindex)

       ..
```

Comment (kmpaul) [reply source/tutorials/yourfirst.rst:1854]: Missed this formatting

File: source/tutorials/yourfirst.rst [hunk 12]
```diff
@@ -1399,3 +1399,1020 @@ repository to GitHub.
...
+       windchill = []
+       for temp, windspeed in zip(data['tempout'], data['windspeed']):
+          windchill.append(compute_windchill(temp, windspeed))
+
+       # Output comparison of data
+       print_comparison('WINDCHILL', data['date'], data['time'], data['windchill'], windchill)
+
+    ..
+
+And for :code:`heatindexcomp.py`\:
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:1973]: Missed this formatting mistake, too...

```suggestion
    And for :code:`heatindexcomp.py`\:
```

File: source/tutorials/yourfirst.rst [hunk 13]
```diff
@@ -1399,3 +1399,1020 @@ repository to GitHub.
...
+   Something new that we have done here is use the \":code:`import ... as ...`\" statement. This essentially allows us to 
+   shorten the name of the module for convenience if it is very long or if we are going to be calling it a lot.
+
+   The symbol :code:`math.e` represents Euler's number (*e*), the base of the natural logarithm. Euler's number (*e*) is an irrational number with infinite decimal places, often approximated as 2.718. 
+   How much more accurate is :code:`math.e` than this approximation?
+   
+   The function :code:`math.log(x, base)` is very useful for computing logarithms in any base - but
+   for some common bases there are separate logarithmic functions. 
+   Try using :code:`log10(x)`\:
+   .. code-block:: python
```
Comment (kmpaul) [inline source/tutorials/yourfirst.rst:2343]: Needs a blank line before the Python code block.

```suggestion

   .. code-block:: python
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
A few formatting errors that are easy to fix.

**Comment by kmpaul** [comment]
Once these formatting errors are fixed, let me know.  I'll go ahead and merge the changes.  Then we should announce the addition of the new content to Xdev and have them walk through it to make sure it works and looks good to them.

**Comment by jukent** [comment]
> Once these formatting errors are fixed, let me know. I'll go ahead and merge the changes. Then we should announce the addition of the new content to Xdev and have them walk through it to make sure it works and looks good to them.

Fixed! Yes it would be great to announce this and get an extra pair of eyes on the website.

**Review by kmpaul — APPROVED** [review]
Let's merge!

**Pull request merged**